### PR TITLE
SCT-833 - Add a new temporary object to the BE response

### PR DESCRIPTION
**Pull request by mapleleafjack**
## Link to JIRA ticket

[Link to ticket](https://hackney.atlassian.net/browse/SCT-833)

## Describe this PR

A "relationship" object has been added to the Personal Relationship response.
This object is very similar to the pre-existing "persons" object, with the difference that it includes the RelationshipID (as "id") together with the PersonID (as "personId").

The RelationshipID is necessary for the relationship deletion request from the Frontend.

### *What changes have we introduced*

Added a second object "relationship" in the Relationship response.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [X] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [X] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

This can be removed after the BE and FE are aligned

### CODE REVIEW

File: SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+namespace SocialCareCaseViewerApi.V1.Domain
+{
+    public class RelatedRelationship
```
Comment (chubberlisk) [inline SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs:3]: I'm sure we could name this better but I can't think of anything! 😅

**Review by chubberlisk — APPROVED** [review]
Looks good to me!

File: SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs (hunk 1 above)
Comment (mapleleafjack) [reply SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs:3]: Maybe "Relationship" or "Relation" can be used, anyways we can change it when we remove the code for "RelatedPerson"

Comment (chubberlisk) [reply SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs:3]: Hmmm, not sure about those as they feel too generic and possibly confusing against other classes. Let's run with this for now and like you said, we can revisit the name.

File: SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        public long Id { get; set; }
+        public long PersonId { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Gender { get; set; }
+        public string IsMainCarer { get; set; }
+        public string IsInformalCarer { get; set; }
+        public string Details { get; set; }
```
Comment (JohnFarrellDev) [inline SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs:12]: Details can be null right?

Can any of the other properties on RelatedRelationship.

I'd advise being explicit about what can and can't be null (https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references)

Comment (mapleleafjack) [reply SocialCareCaseViewerApi/V1/Domain/RelatedRelationship.cs:12]: Yes, Details can be null!

I'll set it nullable and add the #nullable enable annotation right on top of it

**Review by JohnFarrellDev — APPROVED** [review]
:rocket:

**Pull request merged**